src/scalar.rs

```diff
@@ -59,6 +65,22 @@ impl PartialEq for Scalar {
     }
 }
 
+impl Ord for Scalar {
+    fn cmp(&self, other: &Self) -> core::cmp::Ordering {
+        let mut self_bytes = self.to_bytes();
+        let mut other_bytes = other.to_bytes();
+        &self_bytes.reverse();
+        &other_bytes.reverse();
+        self_bytes.cmp(&other_bytes)
+    }
+}
+
+impl PartialOrd for Scalar {
+    fn partial_cmp(&self, other: &Self) -> Option<core::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
```

Comment:
Cryptographic structures, in particular finite fields, do not have a total ordering, (in the "well acktually mathematical sense") so I disagree with exposing `Ord`/`PartialOrd` when they are in the form of `Scalar` since they're expected to behave like finite fields. However, obviously we can totally order them when they're represented in serialized form (as this implementation leverages) assuming they're serialized canonically.

If you need `Ord` for things like a `BTreeMap` (for example) then the end user should convert it into its serialized form and depend on the total ordering of that instead.